src/requireAuth.ts

```diff
@@ -53,9 +53,18 @@ async function autoAuth(options: Options, authScopes: string[]): Promise<void |
     // Within monospace, this a OAuth token for the user, so we make it the active user.
     setActiveAccount(options, {
       user: { email: clientEmail },
-      tokens: { access_token: token },
+      tokens: {
+        access_token: token,
+        expires_at: client.cachedCredential?.credentials.expiry_date,
+      } as TokensWithExpiration, // We need the refresh_token here.
+    });
+    setGlobalDefaultAccount({
+      user: { email: clientEmail },
```

Comment:
Nit: This object is duplicated between the calls to setActive and setGlobal. Consider declaring it once as a variable and reusing.